File: Java_Assignment/Assignment_55/programAQ5.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement55Question5.java
//  Description : Write Java program which accept array from user and display below pattern.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*
Input : 8 9 7 6 4 2 4
Output :

* * * * * * * *
* * * * * *
* * * *
* *
* * * *

*/

import java.util.Scanner;

class MyArray
{
    public void Pattern(int arr[]) 
    {
        int i = 0, j = 0;
        
        for(i = 0; i < arr.length;i++)
        {
            if(arr[i] % 2 == 0)
            {
                for(j = 1;j <= arr[i];j++)
                {
                    System.out.print("* ");
                }
                System.out.println();
            }
        }
    }
}

public class programAQ5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iSize = 0, i = 0;

        System.out.print("Enter size of array : ");
        iSize = sobj.nextInt();

        int arr[] = new int[iSize];

        System.out.println("Enter element:");
        for(i = 0; i < iSize; i++)
        {
            arr[i] = sobj.nextInt();
        }

        MyArray aobj = new MyArray();

        aobj.Pattern(arr);

        sobj.close();
    }   
}

File: CPP_Assignment/Assigment_38/programAQ5.cpp
///////////////////////////////////////////////////////////
//
//  File name : Assignement38Question5.cpp
//  Description : Write generic program to accept N values from user and return smallest value.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//
///////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

template <class T>
T Min(T *arr, int iSize)
{
    T iMin = arr[0];

    int i = 0;

    for(i = 0;i < iSize;i++)
    {
        if(arr[i] < iMin)
        {
            iMin = arr[i];
        }
    }
    return iMin;
}

int main()
{
    int arr[] = {10, 20, 30, 40, 50};
    
    float brr[] = {10.0, 3.7, 9.8, 8.7};

    int iRet = Min(arr, 5);
    
    printf("%d\n", iRet); // 10
    
    float fRet = Min(brr, 4);
    
    printf("%f\n", fRet); // 10.0
    
    return 0;
}